src/neo4j_genai/retrievers/external/weaviate/examples/vector_search.py

```diff
@@ -0,0 +1,46 @@
+#  Copyright (c) "Neo4j"
+#  Neo4j Sweden AB [https://neo4j.com]
+#  #
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#  #
+#      https://www.apache.org/licenses/LICENSE-2.0
+#  #
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from neo4j_genai.retrievers.external.weaviate import WeaviateNeo4jRetriever
+from neo4j import GraphDatabase
+import weaviate
+import sys
+import os
+
+BASE_DIR = os.path.dirname(os.path.abspath(__file__))
+sys.path.append(f"{BASE_DIR}/../../../../../../tests/e2e/weaviate_e2e")
```

Comment:
Why do we need to add this to the path?

Reply:
So I can import the function. Maybe there are other ways?

Reply:
I'm thinking if it make more sense to keep the example outside of `src/` and in `examples/` instead? Imo it should live outside of the source code

Is there a reason the vector needs to sit inside `tests/`? If not we can either duplicate it in both `examples/` and `tests/`. If you think we should still import this function from `tests/` then you can make it cleaner like this:

```python
root_dir = os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
sys.path.insert(0, root_dir)
```

Reply:
The vector is used in the tests, that's why it's there. And I figured we'd reuse it when demonstrating vector search.
Happy to duplicate it though. And will move the examples to the examples folder.

Generally I'm not a fan of grouping files by type but rather by feature (I would move all tests, examples etc right next to the implementation) but I understand that's not very common in the python world.
Long term maintainability would suffer when grouping by type. But That shouldn't block this PR :)

Reply:
Moved around a bit and duplicated the embedding: https://github.com/neo4j/neo4j-genai-python/pull/35/commits/140e610e0875401b145e0f2a58608060ef9a223b